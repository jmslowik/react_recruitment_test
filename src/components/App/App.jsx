import React from 'react';
import ProductsList from '@/containers/ProductsList';
import LocaleContext from '@/contexts/LocaleContext';
import { locale, currency } from '@/config';

const App = () => (
  <LocaleContext.Provider value={{ locale, currency }}>
    <ProductsList />
  </LocaleContext.Provider>
);

export default App;
