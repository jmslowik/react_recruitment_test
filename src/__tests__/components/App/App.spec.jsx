import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import App from '@/components/App/App';

describe('App', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<h3>Lista produktów</h3>)).toBe(true);
  });

  it('should be selectable by class "container"', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.is('.container')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('.container').length).toBe(1);
  });

  it('should render to static HTML', () => {
    const wrapper = render(<App />);
    expect(wrapper.find('li').text()).toEqual('Patelnia, cena: 89,99zł');
  });

  it('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
