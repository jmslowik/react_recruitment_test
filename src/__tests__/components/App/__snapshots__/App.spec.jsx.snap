// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="container"
>
  <h3>
    Lista produktów
  </h3>
  <ul>
    <li
      className="row"
    >
      Patelnia, cena: 89,99zł
    </li>
  </ul>
</div>
`;
